src/django_components/dependencies.py

```diff
@@ -595,7 +755,7 @@ def _postprocess_media_tags(
 
 
 def _prepare_tags_and_urls(
-    data: Iterable[str],
+    data: List[Tuple[str, ScriptType, Optional[str]]],
```

Comment:
Changed how the data is passed to this function, so we don't needlessly generate `<script>` or `<link>` tags for JS / CSS variables, when the component doesn't define any.  But the logic is still the same